bpo-39674: Suggest to test with DeprecationWarning (#18552)
python/cpython: Doc/whatsnew/3.9.rst

.. PEP-sized items next.

You should check for DeprecationWarning in your code
====================================================

When Python 2.7 was still supported, many functions were kept for backward

> compatibility with Python 2.7. With the end of the Python 2.7 support, these

Reviewer: ```suggestion
compatibility with Python 2.7. With the end of Python 2.7 support, these
```

> backward compatibility layers have been reomved or will be removed shortly.

Reviewer: ```suggestion
backward compatibility layers have been removed, or will be removed soon.
```

> Most of them emitted :exc:`DeprecationWarning` warning for several years. For

Reviewer: ```suggestion
Most of them emitted a :exc:`DeprecationWarning` warning for several years. For
```

example, using ``collections.Mapping`` instead of ``collections.abc.Mapping``

> emits a :exc:`DeprecationWarning` since Python 3.3 released in 2012.

Reviewer: ```suggestion
emits a :exc:`DeprecationWarning` since Python 3.3, released in 2012.
```

> Test your application with :option:`-W` ``default`` command line option to see

Reviewer: ```suggestion
Test your application with the :option:`-W` ``default`` command-line option to see
```

:exc:`DeprecationWarning` and :exc:`PendingDeprecationWarning`, or even with
:option:`-W` ``error`` to treat them as errors. :ref:`Warnings Filter

> <warning-filter>` can be used to ignore warnings from third party code.

Reviewer: ```suggestion
<warning-filter>` can be used to ignore warnings from third-party code.
```

:class:`collections.abc.Mapping`, are kept for one last release for backward
compatibility. They will be removed from Python 3.10.

More generally, try to run your tests in the :ref:`Python Development Mode

> <devmode>` which helps to prepare your code to make it compatiblity with the

Reviewer: ```suggestion
<devmode>` which helps to prepare your code to make it compatible with the
```
